### ISSUE TITLE
setData does not work with pie charts, again.

### ISSUE THREAD

**Issue by chriscareycode** [highcharts/highcharts#5209]
When calling setData() on pie charts, the data is not updated in the pie chart unless the number of pie slices changes.

With HighCharts 4.1.4 the bug does not exist.
Somewhere between 4.1.4 to 4.1.10 the bug was introduced. The bug still exists on 4.2.3.

Referencing this old bug: 
https://github.com/highcharts/highcharts/issues/542

**Comment by TorsteinHonsi** [comment]
Thanks for reporting! It seems like you forgot to provide a live demo, so I tried to set up one. I tried using both numeric and object point configurations, but they both work well: http://jsfiddle.net/highcharts/5azwzo4c/.

Please do the changes you need in order to reproduce the issue.

**Comment by chriscareycode** [comment]
I am using Highcharts within an EmberJS component. Your example did provide the ability to determine what is going wrong. I set up the chart with my exact initialization code and the problem was not happening in jsfiddle. So I took a look closer at the data. EmberJS is doing something strange to the data array of objects which is causing the problem. Before calling setData(), if I create a new array of objects by mapping the data, then assigning that new array in the setData(), the problem disappears. 

```
let newData = data.map(function(d) {
      return {
        name: d.name,
        y: d.y
      };
    });
```
Even though the problem does not manifest itself on 4.1.4 but only started on newer versions I think the problem is with EmberJS and whatever optimization it is doing to the data object I'm passing it. Thank you for your help and sorry to waste your time!
